base/range.jl

```diff
@@ -786,3 +786,5 @@ end
 
 in{T<:Integer}(x, r::Range{T}) = isinteger(x) && !isempty(r) && x>=minimum(r) && x<=maximum(r) && (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)
 in(x::Char, r::Range{Char}) = !isempty(r) && x >= minimum(r) && x <= maximum(r) && (mod(Int(x) - Int(first(r)), step(r)) == 0)
+
+unique(x::Range) = x
```

Comment:
This was brought up in #14649. I had thought that the above behavior behavior was supposed to be a bug but reading back on the comments it looks like its still ambiguous. This PR has lingered on a bit long, so I modified the method to accommodate either case. WRT to the larger issue of confusion about range with zero step sizes see #15218.